src/modules-by-example.adoc

```diff
@@ -0,0 +1,320 @@
+  ],
+  install_symbols: [ local ]
+}
+----
+
+The definition of the local symbol table has been refactored into two parts. First, the list of
+symbols is expressed inside a module named `local`. Then, the symbols from that module are
+installed to form the new local symbol table. Compared to the behavior of `$ion_symbol_table`,
+this is akin to defining a named symbol table “inline” to hold local symbols, then defining the
+local symbol table only via `imports` and no `symbols` field.
+
+Let's look more closely at the definition of `local`:
+
+[{nrm}]
+----
+(*module* local
+  (*symbols* ["s1", "s2"]))
+----
+
+The `*module*` keyword starts an S-expression that defines a new module with the given name.
+The `*symbols*` keyword starts a subform that defines the module's exported symbol table.
+This clause expects a list of strings, and both the syntax and semantics are the same as the
+`symbols` field of `$ion_shared_symbol_table`.
+
+Once this module is defined, we can _install_ it into the local symbol table:
+
+[{nrm}]
+----
+  install_symbols: [ local ]
+----
+
+This field expects a list of symbols that match names declared in the `modules` field.  The
+resulting local symbol table is simply the concatenation of the exported symbol tables of those
+modules.  This works the same way as the `imports` field of `$ion_symbol_table`.
+
+
+=== Importing Symbols
+
+Given the equivalencies above, we could perform a naive round-trip of the preceding 1.1 document
+back to 1.0. First, turn the `local` module into the equivalent shared symbol table:
+
+[{nrm}]
+----
+**$ion_shared_symbol_table**::{
+  name: "local",
+  version: 1,
+  symbols: ["s1", "s2"]
+}
+----
+
+Then translate `install_symbols:[local]` into its 1.0 equivalent:
+
+[{nrm}]
+----
+*$ion_1_0*
+**$ion_symbol_table**::{
+  imports: [{ name: "local", version: 1, max_id: 2 }]
+}
+----
+
+NOTE:  Even ignoring Ion 1.1, this is how you would hoist local symbols into a
+new shared symbol table.
+
+The latter imports-only document has this 1.1 equivalent:
+
+
+[{nrm}]
+----
+*$ion_1_1*
+**$ion_encoding**::{
+  modules: [
+    (*import* local "local" 1 2)
+  ],
+  install_symbols: [ local ]
+}
+----
+
+Here we see a new form inside the `modules` field that imports a module into the encoding
+environment and assigns it a name.
+The `*import*` keyword starts an S-expression that expects three or four arguments. The first is
+a symbolic name that we can use later to refer to the imported module.  The remaining arguments
+are effectively the `name`, `version` and `max_id` fields of the 1.0 `imports` struct.
+
+TIP: From the perspective of Ion 1.1, shared symbol tables _are_ encoding modules.
+
+
+=== Declaring Multiple Modules
+
+Within a directive's `modules` field, both `*module*` and `*import*` clauses have the same
+ultimate effect, declaring a name->module mapping for the encoding environment.
+Let’s look at a scenario with both imported and locally-defined symbols:
```

Comment:
These paragraph seems a little out of place--I was confused because of the Ion 1.0 example below.  Maybe set up the 1.0 example then move this down.